File: typescript/src/databases/es.ts
import { Client } from "@elastic/elasticsearch";

export default (): Client => {
  const client = new Client({
    node: process.env.ES_URL,
  });

  client.ping().catch((err) => {
    console.error(`ElasticSearch connection error.`, err.message);

    // exit only in prod
    if (process.env.NODE_ENV === "production") {
      console.error(err);
      process.exit(1);
    }
  });

  return client;
};

File: go/internal/database/es.go
package database

import (
	"fancy-todo/internal/config"
	"os"

	"github.com/elastic/go-elasticsearch/v7"
)

func NewEs(env *config.Env) (*elasticsearch.Client, error) {
	esCfg := elasticsearch.Config{
		Addresses: []string{
			env.EsUrl,
		},
	}

	es, err := elasticsearch.NewClient(esCfg)
	if err != nil {
		 return nil, err
	}

	if (os.Getenv("GO_ENV") == "production") {
		_, err := es.Ping()
		if err != nil {
			return nil, err
		}
	}

	return es, nil
}